authenticate/authenticate.go

```diff
@@ -227,3 +223,23 @@ func New(opts config.Options) (*Authenticate, error) {
 
 	return a, nil
 }
+
+func (a *Authenticate) setAdminUsers(opts *config.Options) {
+	a.administrator = make(map[string]struct{}, len(opts.Administrators))
+	for _, admin := range opts.Administrators {
+		a.administrator[admin] = struct{}{}
+	}
+}
+
+// UpdateOptions implements the OptionsUpdater interface and updates internal
+// structures based on config.Options
+func (a *Authenticate) UpdateOptions(opts config.Options) error {
+	if a == nil {
+		return nil
+	}
+
+	log.Info().Str("checksum", fmt.Sprintf("%x", opts.Checksum())).Msg("authenticate: updating options")
+	a.setAdminUsers(&opts)
```

Comment:
this isn't thread safe. If someone attempts to use `administrator` while it is being written it will cause a panic.

Reply:
I don't think it will panic, but can give you wrong information. I will a a lock